partner_default_delivery_carrier/models/res_config_settings.py

```diff
@@ -0,0 +1,32 @@
+# Copyright 2021 Camptocamp
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class ResConfigSettings(models.TransientModel):
+    _inherit = "res.config.settings"
+
+    partner_default_delivery_carrier_id = fields.Many2one("delivery.carrier")
+
+    def get_values(self):
```

Comment:
This is not the good implementation in this version.

You should declare the field also on res.company. And here, change the field as related as : `related=company_id.partner_default_delivery_carrier_id`.